xbmc/filesystem/CacheStrategy.cpp

```diff
@@ -329,31 +329,28 @@ int64_t CDoubleCache::Seek(int64_t iFilePosition)
 
 bool CDoubleCache::Reset(int64_t iSourcePosition, bool clearAnyway /* = false */)
 {
-  if (!clearAnyway && m_pCache->IsCachedPosition(iSourcePosition)
-      && (!m_pCacheOld || !m_pCacheOld->IsCachedPosition(iSourcePosition)
-          || m_pCache->CachedDataEndPos() >= m_pCacheOld->CachedDataEndPos()))
+  if (clearAnyway || !m_pCache->IsCachedPosition(iSourcePosition))
```

Comment:
@wsnipex: What's your idea about the (old) logic here? It used to consider which cache had the "highest" end position but I think that logic is wrong/flawed but I also have doubts that just swapping in case of a seek may not be the best solution for all possible scenarios.